tasks/terraform-cli/src/context/azdo-task-context.ts

```diff
@@ -3,13 +3,15 @@ import * as tasks from 'azure-pipelines-task-lib/task';
 import { BackendTypes } from "../backends";
 
 const isCommand = (...commands: string[]) => (ctx: ITaskContext) => commands.includes(ctx.name);
+const isCommandWithSubCommand = (commands: string[], subCommands: string[], subCommand: (ctx: ITaskContext) => string) => (ctx: ITaskContext) => commands.includes(ctx.name) && subCommands.includes(subCommand(ctx));

```

Comment:
@charleszipp due to having separate `subCommand` for `workspace` and  `state` I've added a third attribute to get the proper value, not super fan of this tbh, have another idea?

Reply:
I think this is fine for now. The only other approach I can think of is updating `isCommand` and adding `subCommands` as an optional second parameter. To make it optional, you can use the spread operator (similar to `commands` param in `isCommand`). This will make the evaluation more complex however as you will have to check if subCommands has any elements first then runs includes.

I think what you currently have is more readable and simpler to understand than that alternative so I would keep it for now.